base/loading.jl

```diff
@@ -1093,6 +1094,16 @@ function _require_prelocked(uuidkey::PkgId)
         error("package `$(uuidkey.name)` did not define the expected \
               module `$(uuidkey.name)`, check for typos in package module name")
     end
+    pkgorig = get(pkgorigins, uuidkey, nothing)
+    new_path = locate_package(uuidkey)
+    if pkgorig !== nothing && pkgorig.path !== nothing &&
+            !samefile(fixup_stdlib_path(pkgorig.path), new_path) && uuidkey.uuid ∉ already_warned_path_change_pkgs
+        @warn "Package $(uuidkey.name) already loaded from path $(repr(pkgorig.path)), \
+               now attempted to be loaded from $(repr(new_path)). This might indicate a \
+               change of environment between package loads and can mean that incompatible \
+               packages have been loaded."
```

Comment:
```suggestion
               packages have been loaded.
               If you are seeing this because of packages loaded in your startup.jl you may want to start julia directly in
               the project you are working from via the `--project` arg, which will prefer versions from that environment"
```